src/org/jetbrains/java/decompiler/modules/decompiler/stats/IfStatement.java

```diff
@@ -489,4 +489,44 @@ public boolean match(MatchNode matchNode, MatchEngine engine) {
     Integer type = (Integer) matchNode.getRuleValue(MatchProperties.STATEMENT_IFTYPE);
     return type == null || this.iftype == type;
   }
+
+  public boolean checkInversion() {
```

Comment:
I think the name needs to be more descriptive, as `check` implies to me that it's pure. Perhaps something like `tryInvert`?

Reply:
Mmh, It only inverts when it "needs" to. `tryInvert` sounds like it will invert if possible, even when it isn't needed.

Reply:
what about something like `fixInversion`